Functions/Import/xASL_imp_StudyParPriority.m

```diff
@@ -0,0 +1,90 @@
+function studyParSpecific = xASL_imp_StudyParPriority(studyParFull, subjectName, sessionName, runName)
+%xASL_imp_StudyParPriority Takes the studyParFull a prioritizes based on the current subject/session/run
+%
+% FORMAT: studyParSpecific = xASL_imp_StudyParPriority(studyParFull[, subjectName, sessionName, runName])
+%
+% INPUT:
+%   studyParFull - StudyPar possibly containing several contexts (REQUIRED, STRUCT)
+%   subjectName  - Name of the current subject (OPTIONAL, DEFAULT = '')
+%   sessionName  - Name of the current session (OPTIONAL, DEFAULT = '')
+%   runName      - Name of the current run (OPTIONAL, DEFAULT = '')
+%
+% OUTPUT:
+%   studyParSpecific - Resolved studyPar for a specific subject/session/run
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:    Takes studyPar with possibly several contexts and resolves the priority and saves the individual studyPar.
+%                 First context has the lowest priority, following ones are more important and review previous variables. 
+%                 The alias hierarchy within each context contains a list of regexp to be matched against subject/session/run.
+%                 If subject doesn't exist or regexp doesn't exist or aliasHierarchy doesn't exist, it allows it.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:        studyPar = xASL_imp_StudyParPriority(studyParFull, 'sub-01', '1', '2')
+% __________________________________
+% Copyright 2015-2022 ExploreASL
+
+% Check existence of studyParFull and return if single context
+if nargin < 1 || isempty(studyParFull)
+	studyParSpecific = struct();
+	return;
+end
+
+% Do nothing for a single-context studyPar
+if ~isfield(studyParFull,'StudyPars')
+	studyParSpecific = studyParFull;
+	return;
+end
+
+% Initialize empty studyPar and then go through different contexts
+studyParSpecific = struct();
+
+% At each step, check for all conditions and set a flag for overwrite (first step having the lowest priority
+for iContext = 1:length(studyParFull.StudyPars)
+	bOverwrite = 1;
+	
+	% Checks for fields SubjectRegexp, VisitRegexp, SessionRegexp - if missing or empty, then take this context into account
```

Comment:
Yes. I don't see that huge difference between `studyParAll` and `studyParFull`, and `studyParSpecific` and `studyParIndivSubjVisitSess` to justify the change ;)

Reply:
OK but then you need to explain in the header what `studyParSpecific` means. Sorry for being annoying, again my mission is to make reading code easier for the non-educated programmer ;)

Reply:
It is already explained in the function header. Note that `studyParIndivSubjVisitSess` isn't any better or self-explanatory. We can discuss if `indiv` or `specific` is a better keyword though.

Reply:
Well `specific`is not self-explanatory. Specific to what? Specific to the labeling? The readout? This vendor?

So I'm also fine with `studyParSpecificSubjVisitSess`. it would be nice to also explain in the subject header such variable names that are not required for input output but are used in the function only.

You can choose and resolve this conversation.

Reply:
ok - fine with that suggestion. That variable is explained in the header of the called function ;)